src/main/java/com/denizenscript/ddiscordbot/commands/DiscordBanCommand.java

```diff
@@ -0,0 +1,116 @@
+        setRequiredArguments(3, 6);
+        isProcedural = false;
+        autoCompile();
+    }
+
+    // <--[command]
+    // @Name discordban
+    // @Syntax discordban [id:<id>] ({add}/remove) [user:<user>] [group:<group>] (reason:<reason>) (deletion_timeframe:<time>/{0s})
+    // @Required 3
+    // @Maximum 6
+    // @Short Bans or unbans a member from a group.
+    // @Plugin dDiscordBot
+    // @Guide https://guide.denizenscript.com/guides/expanding/ddiscordbot.html
+    // @Group external
+    //
+    // @Description
+    // Bans or unbans a member from a group.
+    //
+    // To ban a user, use the "add" argument. To unban a user, use the "remove" argument.
+    // The group is required for both "add" and "remove" arguments, but "reason" can only be used with "add".
+    // Reasons show up in the group's Audit Logs.
+    //
+    // The "deletion_timeframe" argument will, if set, delete all messages sent by the user being banned within the timeframe given.
+    // The timeframe defaults to 0 seconds, which will not delete any messages. The timeframe cannot be greater than 7 days.
+    // This argument can only be used when adding a ban using the "add" argument, although it is not required.
+    //
+    // The command should usually be ~waited for. See <@link language ~waitable>.
+    //
+    // @Tags
+    // <entry[saveName].user> returns the DiscordUserTag of the user being banned or unbanned.
+    // <DiscordUserTag.is_banned[<group>]> returns if the user is banned from a certain group.
+    // <DiscordGroupTag.banned_members> returns a list of all banned members in a group.
+    //
+    // @Usage
+    // # Bans a user with a reason and deletes all messages sent by the user in the past 2 hours.
+    // - ~discordban id:mybot add user:<[user]> group:<[group]> "reason:Was being mean!" deletion_timeframe:2h
+    //
+    // @Usage
+    // # Bans a user with but does not delete any messages sent and does not have a reason.
+    // - ~discordban id:mybot add user:<[user]> group:<[group]>
+    //
+    // @Usage
+    // # Unbans a user.
+    // - ~discordban id:mybot remove user:<[user]> group:<[group]>
+    // -->
+
+    public enum DiscordBanInstruction { ADD, REMOVE }
+
+    public static void autoExecute(ScriptEntry scriptEntry,
+                                   @ArgPrefixed @ArgName("id") DiscordBotTag bot,
+                                   @ArgName("instruction") DiscordBanInstruction instruction,
+                                   @ArgPrefixed @ArgName("user") DiscordUserTag user,
+                                   @ArgPrefixed @ArgName("group") DiscordGroupTag group,
+                                   @ArgPrefixed @ArgDefaultNull @ArgName("reason") String reason,
+                                   @ArgPrefixed @ArgDefaultNull @ArgName("deletion_timeframe") DurationTag deletionTimeframe) {
+        if (group != null && group.bot == null) {
+            group.bot = bot.bot;

```

Comment:
Not your fault, this was wrong in pre-existing code that you referenced, but change this to:
`group = new DiscordGroupTag(bot.bot, group.guild_id);`